### PR TITLE
Enrolment functionality and associated tests and docs

### PR DESCRIPTION
**Pull request by chrift**

### CODE REVIEW

File: lib/methods/dynamics/webApi/read.js [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+
+  readServiceEnrolment.parseResponse = (enrolment) => {
+    const response = {
+      roles: [],
+      mappings: []
+    }
+
+    enrolment.value.forEach(enrol => {
+      response.roles.push(`${enrol._defra_organisation_value}:${enrol._defra_servicerole_value}:${enrol.defra_enrolmentstatus}`)
+
```
Comment (brendanarnold) [inline lib/methods/dynamics/webApi/read.js]: Blank lines?

File: lib/internals/index.js [hunk 2]
```diff
@@ -17,7 +19,15 @@ module.exports = ({
       server,
       cache,
       config
-    })
+    }),
+    dynamics: {
+      ...require('./dynamics')({
```
[the remark this answers is not in the document]
Comment (chrift) [reply lib/internals/index.js]: The dynamics module returns a function which relies on the input parameters being passed into it here.
Mappings just exports an object

File: lib/methods/dynamics/webApi/read.js [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+      params['$filter'] += ` and (`
+
+      params['$filter'] += serviceRoleIds.map(id => `_defra_servicerole_value eq ${id}`).join(' or ')
+
+      params['$filter'] += ')'
+    }
+
+    if (accountIds) {
+      params['$filter'] += ` and (`
+
```
[the remark this answers is not in the document]
Comment (chrift) [reply lib/methods/dynamics/webApi/read.js]: Have removed these

File: lib/methods/dynamics/webApi/read.js (hunk 1 above)
Comment (chrift) [reply lib/methods/dynamics/webApi/read.js]: Just indicating separating between pushing to roles and mappings

**Pull request merged**